gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -138,9 +159,11 @@ void onSignedOut() {
     navigator.navigate(SignInFragmentDirections.showSignInScreen());
   }
 
-  void onSignedIn() {
-    hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+  private Completable onSignedIn(User user) {
+    return userRepository
+        .saveUser(user)
```

Comment:
Can we move the user creation under if block `onSignInStateChange` and call `onSignedIn` from there?